include/ginkgo/core/base/range_accessors.hpp

```diff
@@ -186,6 +216,647 @@ class row_major {
+     *
+     * @param storage  pointer to the block of memory containing the storage
+     * @param size  multidimensional size of the memory
+     */
+    GKO_ATTRIBUTES constexpr reduced_row_major(dim<dimensionality> size,
+                                               storage_type *storage)
+        : reduced_row_major{size, storage,
+                            helper::compute_stride_array<size_type>(size)}
+    {}
+
+    /**
+     * Creates an empty accessor (pointing nowhere with an empty size)
+     */
+    GKO_ATTRIBUTES constexpr reduced_row_major()
+        : reduced_row_major{{0, 0, 0}, nullptr}
+    {}
+
+public:
+    /**
+     * Creates a reduced_row_major range which contains a read-only version of
+     * the current accessor.
+     *
+     * @returns  a reduced_row_major major range which is read-only.
+     */
+    GKO_ATTRIBUTES GKO_INLINE constexpr range<const_accessor> to_const() const
+    {
+        return range<const_accessor>{size_, storage_, stride_};
+    }
+
+    /**
+     * Returns the length in dimension `dimension`.
+     *
+     * @param dimension  a dimension index
+     *
+     * @returns  length in dimension `dimension`
+     */
+    GKO_ATTRIBUTES GKO_INLINE constexpr size_type length(
+        size_type dimension) const
+    {
+        return dimension < dimensionality ? size_[dimension] : 1;
+    }
+
+    /**
+     * Copies data from another accessor
+     *
+     * @warning Do not use this function since it is not optimized for a
+     *          specific executor. It will always be performed sequentially.
+     *          Please write an optimized version (adjusted to the architecture)
+     *          by iterating through the values yourself.
+     *
+     * @tparam OtherAccessor  type of the other accessor
+     *
+     * @param other  other accessor
+     */
+    template <typename OtherAccessor>
+    GKO_ATTRIBUTES void copy_from(const OtherAccessor &other) const
+    {
+        helper::multidim_for_each(size_, [&](auto... indices) {
+            (*this)(indices...) = other(indices...);
+        });
+    }
+
+    /**
+     * Returns the stored value for the given indices. If the storage is const,
+     * a value is returned, otherwise, a reference is returned.
+     *
+     * @param indices  indices which value is supposed to access
+     *
+     * @returns  the stored value if the accessor is const (if the storage type
+     *           is const), or a reference if the accessor is non-const
+     */
+    template <typename... Indices>
+    GKO_ATTRIBUTES GKO_INLINE constexpr std::enable_if_t<
+        helper::are_all_integral<Indices...>::value,
+        std::conditional_t<is_const, arithmetic_type, reference_type>>
+    operator()(Indices &&... indices) const
+    {
+        return reference_type{storage_ +
+                              compute_index(std::forward<Indices>(indices)...)};
+    }
+
+    /**
+     * Returns a sub-range spinning the current range (x1_span, x2_span, ...)
```

Comment:
```suggestion
     * Returns a sub-range spanning the current range (x1_span, x2_span, ...)
```
? spinning looks like you shuffle them

Reply:
Yes, good find, it was a typo.